src/core/thread/mle.cpp

```diff
@@ -520,6 +520,7 @@ ThreadError Mle::SetRloc16(uint16_t aRloc16)
     }
 
     mMac.SetShortAddress(aRloc16);
+    mNetif.GetNetworkDataLocal().SendServerDataNotification();
```

Comment:
Task might be not required if `SendServerDataNotification()` is called in `UpdateRoutes()`.
One problem is for pre-configured `rsn` child border server,  there might be no `UpdateRoutes()` for `rsn` device.